### PR TITLE
Fix existing OSM layers and improve bus lane detection

### PR DESCRIPTION
**Pull request by dabreegster**
While regenerating all the OSM layers with latest data, I found a problems from prior refactors. Mainly, the "source layer name" used in pmtiles is based on the input geojson file, so it was called "extract" incorrectly. I fixed this, and also tried to improve the helper functions.

And mainly, I added in some missing bus lane cases I noticed, like https://www.openstreetmap.org/way/968336703

### CODE REVIEW

File: layers/generate_layers.py [hunk 1]
```diff
@@ -99,8 +99,8 @@ def main():
         )
 
 
-# Extract `{tagPartOne}={tagPartTwo}` polygons from OSM, and only keep a name attribute.
-def generatePolygonLayer(osm_input, tagPartOne, tagPartTwo, filename):
+# Extract polygons from OSM using a tag filter, and only keep a name attribute.
```
Comment (dabreegster) [inline layers/generate_layers.py:102]: I don't really think it's helpful to split into two parameters. Whoever's calling this has to go read the osmium docs to figure out the format anyway

File: layers/generate_layers.py [hunk 2]
```diff
@@ -659,21 +590,29 @@ def roadHasBusLane(tags):
...
+
     # We're not handling highway=busway or other cases for service roads
     # designed exclusively for buses, because they're not intended for cyclists
     # or any other users.
 
     return False
 
 
-def generateGeojsonFromOSMFile(osmFilePath, outputFilepath):
+def convertPbfToGeoJson(pbfPath, geojsonPath, geometryType):
```
Comment (dabreegster) [inline layers/generate_layers.py:607]: We always want to be explicit about exporting points, linestrings, or polygons, so add it as a mandatory param

File: layers/generate_layers.py [hunk 3]
```diff
@@ -689,12 +628,9 @@ def run(args):
     subprocess.run(args, check=True)
 
 
-# For each GeoJSON feature, keep only the name attribute. Overwrites the given file.
```
Comment (dabreegster) [inline layers/generate_layers.py:692]: Just had one caller and the one-line call seems more clear anyway

File: layers/generate_layers.py [hunk 4]
```diff
@@ -689,12 +628,9 @@ def run(args):
...
-def removeNonNameProperties(path):
-    cleanUpGeojson(path, ["name"])
-
-
-def cleanUpGeojson(path, propertiesToKeep, addIds=False):
+# Adds numeric IDs to every feature, trims coordinate precision, and only keeps
+# the specified properties. Overwrites the file.
+def cleanUpGeojson(path, propertiesToKeep):
```
Comment (dabreegster) [inline layers/generate_layers.py:633]: Every caller wants to add IDs, so removed the option

File: layers/generate_layers.py (hunk 1 above)
Comment (Pete-Y-CS) [reply layers/generate_layers.py:102]: I agree, nice change

**Review by Pete-Y-CS — APPROVED** [review]

**Pull request merged**